src/Core/Commander/ManagerCommands.js

```diff
@@ -71,43 +79,80 @@ define('Core/Commander/ManagerCommands', [
             return this.commandsLength()===0;
         };
 
-        ManagerCommands.prototype.runAllCommands = function() {
-
-
-            if (this.commandsLength() === 0)
-            {
-                return Promise.resolve(0);
-            }
-
-            return Promise.all(this.arrayDeQueue(16))
-                .then(function () {
-                    // if (this.commandsLength() <= 16)
-                    this.scene.wait(1);
-                    // else
-                    //     this.scene.renderScene3D();
-                    return this.runAllCommands();
-                }.bind(this));
+        ManagerCommands.prototype.resetExecutedCommandsCount = function() {
+            this.counters.executedCommands = 0;
         };
 
-        ManagerCommands.prototype.arrayDeQueue = function(number) {
-
-            var nT = number === undefined ? this.queueAsync.length : number;
-
-            var arrayTasks = [];
+        ManagerCommands.prototype.executeCommands = function() {
+
+            var command;
+            var launchCommand = function(cmd) {
+                this.counters.runningCommands++;
+                var layer = cmd.layer;
+                var host;
+                if(layer.url) {
+                    host = new URL(layer.url).host;
+                    if(this.counters.hostCommands[host] === undefined) {
+                        this.counters.hostCommands[host] = 1;
+                    } else {
+                        this.counters.hostCommands[host]++;
+                    }
+                }
 
-            while (this.queueAsync.length > 0 && arrayTasks.length < nT) {
-                var command = this.deQueue();
+                var commandEnd = function() {
+                    this.counters.runningCommands--;
+                    this.counters.executedCommands++;
+                    if(host) {
+                        this.counters.hostCommands[host]--;
+                        if(this.counters.hostCommands[host] === 0) {
+                            this.counters.hostCommands[host] = undefined;
+                        }
+                    }
+                    this.executeCommands();
+                }.bind(this);
+
+                var p = this.providers[layer.protocol];
+                if(p) {
+                    var promise = p.executeCommand(cmd);
+                    if(promise) {
+                        promise.then(function() {
+                            commandEnd();
+                        }.bind(this));
+                    } else {
+                        commandEnd();
+                    }
+                } else {
+                    commandEnd();
+                }
+            }.bind(this);
+
+            var skippedCommands = [];
```

Comment:
We could have a sorted commands queue per host so we don't have to implement this skipping mechanism.
Or if we believe that having a single queue is useful (for global sorting for instance) then we could consider changing the underlying storage to better support our use case.